### ISSUE TITLE
Ensure Tag Manager update are run on updates

### ISSUE THREAD

**Issue by nicomegadrive** [piwik/piwik#19887]
Hi,

I'm currently having a problem showing the list of my tags in my tag manager, there is a javascript error and nothing is showing.

Here's the error I got :

TypeError: Cannot read properties of undefined (reading 'length')
    at tagManagerHelper.truncateText (index.php?module=Proxy&action=getCoreJs&cb=dd3ee26d5cec01ecbb10a761a42c944f:2467:308)
    at Proxy.truncateText (index.php?module=Proxy&action=getUmdJs&chunk=2&cb=dd3ee26d5cec01ecbb10a761a42c944f:100:8430)
    at index.php?module=Proxy&action=getUmdJs&chunk=2&cb=dd3ee26d5cec01ecbb10a761a42c944f:100:3023
    at e.renderList (index.php?module=Proxy&action=getCoreJs&cb=dd3ee26d5cec01ecbb10a761a42c944f:41:121877)
    at index.php?module=Proxy&action=getUmdJs&chunk=2&cb=dd3ee26d5cec01ecbb10a761a42c944f:100:2684
    at o (index.php?module=Proxy&action=getCoreJs&cb=dd3ee26d5cec01ecbb10a761a42c944f:41:14442)
    at e.renderSlot (index.php?module=Proxy&action=getCoreJs&cb=dd3ee26d5cec01ecbb10a761a42c944f:41:122378)
    at Proxy.Qn (index.php?module=Proxy&action=getUmdJs&chunk=0&cb=dd3ee26d5cec01ecbb10a761a42c944f:275:966)
    at Dt (index.php?module=Proxy&action=getCoreJs&cb=dd3ee26d5cec01ecbb10a761a42c944f:41:14750)
    at fe.fn (index.php?module=Proxy&action=getCoreJs&cb=dd3ee26d5cec01ecbb10a761a42c944f:41:40794)
(anonymous) @ index.php?module=Proxy&action=getCoreJs&cb=dd3ee26d5cec01ecbb10a761a42c944f:41

Before it happened I set a tag for an A/B test and maybe the script I copy / pasted in one of the tag was too big.

Thanks in advance for your help.

**Comment by thesakshidiggikar** [comment]
hello! I would like to work on this issue could you please assign me this work, please...

**Comment by MrCodeB2** [comment]
Maybe this issue is connected to mine which occurs since the update from 4.12.0 to 4.12.1

Within my tag manager container i can only see items in the Dashboard. When trying to open triggers or tags i get an empty view back (see screenshots).

When trying to create a new trigger (-> Dashboard-> Create new trigger) i get an error when trying to save the trigger saying
SQLSTATE[42S22]: Column not found: 1054 Unknown column 'description' in 'field list'

Seems like a bug in the database upgrade?

![firefox_MptdbsI4nA](https://user-images.githubusercontent.com/56112299/196906324-83b684a9-8fee-4624-b1f0-677358b79b1e.png)
![firefox_Vb0oxjidRs](https://user-images.githubusercontent.com/56112299/196906336-ec6451dc-c110-4142-9432-0eaf791a8564.png)
![firefox_tBJtAVoYNs](https://user-images.githubusercontent.com/56112299/196906370-d40b0ca8-5a8a-4c18-9708-336bd01eeb00.png)

**Comment by AltamashShaikh** [comment]
@nicoco59 @MrCodeB2 
Can you check this [issue](https://github.com/matomo-org/tag-manager/issues/557) ?

**Comment by nicomegadrive** [comment]
I asked my server host to solve the issue with "SQLSTATE[42S22]: Column not found: 1054 Unknown column 'description' in 'field list'" as I got it as well.

It seems I can update tags again now, but I still can't see the list of tags (triggers as well btw), they're investing the problem right now, and of course there is no fallback if I disable javascript.

I'm doing two big A / B tests at the moment, and the HTML code that was generated was quite big, so I was wondering if it's not what caused the error in the first place.

That's quite problematic because I have to stop one A/B test very soon and I just can't from the Tag Manager.

**Comment by MrCodeB2** [comment]
@AltamashShaikh thank you, worked like a charm

**Comment by nicomegadrive** [comment]
The fix above didn't help us, however we found another way to fix it :

After investigation, a column "description" has been added with 3 tables since the last update (4.12.1).
However by looking at the tables, a bug occured during the creation.

The column "description" should have been added after the the column "name" and the content shouldn't be "NULL".
Instead of this, the column "description" has been added as the last column and the content of every entry was "NULL".

Here are the steps to solve this issue :

Backup the tables which are causing the issues
Delete the column "description"
Recreate the column "description"

It's working and I can see the tags again in the Tag Manager.
